### ISSUE TITLE
ignore-missing-import option does not work on false positive "Library stubs not installed for X"

### ISSUE THREAD

**Issue by kbrose** [python/mypy#10601]
**Bug Report**

Sometimes, multiple packages on pypi will share a namespace once installed. One example pair is [`google-cloud-storage`](https://pypi.org/project/google-cloud-storage/) and [`protobuf`](https://pypi.org/project/protobuf/). If you pip install both of these packages, you can see that they both get installed under a top level `google/` folder:

```python
>>> from google.cloud import storage
>>> from google.protobuf import message
>>> storage.__file__
'.../python3.9/site-packages/google/cloud/storage/__init__.py'
>>> message.__file__
'.../python3.9/site-packages/google/protobuf/message.py'
```

This is a problem when one of the packages has a hit in typeshed, but the other does not. In this case, there is a `types-protobuf` package, but not a `types-google-cloud-storage` package. Running `mypy` will _always_ error, even when `--ignore-missing-imports` is set. I believe this happens because it _thinks_ that a known library stub is missing (probably because of the shared `google` namespace), but this is not the case.

Further, the error message Hints at an incorrect solution:

```
note: Hint: "python3 -m pip install types-protobuf"
```

This message shows up even when `types-protobuf` is already installed.

**To Reproduce**

Create two files, `requirements.txt` and `repro.py`:

```bash
# requirements.txt
protobuf
google-cloud-storage
types-protobuf
mypy
```

```python
# repro.py
from google.cloud import storage
from google.protobuf import message
```

Install the packages: `pip install -r requirements.txt`

Run `mypy`:

```
$ mypy .
repro.py:1: error: Library stubs not installed for "google.cloud" (or incompatible with Python 3.9)
repro.py:1: note: Hint: "python3 -m pip install types-protobuf"
repro.py:1: note: (or run "mypy --install-types" to install all missing stub packages)
repro.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```
Unfortunately, `--ignore-missing-imports` does not get rid of the error.
```
$ mypy . --ignore-missing-imports
repro.py:1: error: Library stubs not installed for "google.cloud" (or incompatible with Python 3.9)
repro.py:1: note: Hint: "python3 -m pip install types-protobuf"
repro.py:1: note: (or run "mypy --install-types" to install all missing stub packages)
repro.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

I may be missing something, but as far as I can tell there is no way to force this error to go away. Since it _always_ errors out, this makes it very hard to typecheck my code, especially in an automated CICD pipeline.

Also note that the hint is not helpful, as it suggests installing a package that is already installed!

```
$ python3 -m pip install types-protobuf
Requirement already satisfied: types-protobuf in .../python3.9/site-packages (0.1.10)
Requirement already satisfied: types-futures in .../python3.9/site-packages (from types-protobuf) (0.1.3)
```

**Expected Behavior**

* I would expect `--ignore-missing-imports` to get rid of this error.
* The suggested hint is wrong, and I would expect it to not be shown.

**Actual Behavior**

* No way to get rid of this "error".
* Incorrect Hint is shown.

**Your Environment**

- Mypy version used: 0.900.
- Mypy command-line flags: see above
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.9.1
- Operating system and version: macos

**Comment by alexanderdamiani** [comment]
I am encountering the same issue.

**My environment**
* OS: Windows 10
* Python version: 3.8.8
* mypy version: 0.901

---

I was not seeing the issue in mypy version 0.812.

**Comment by kbrose** [comment]
It seems related to [`FindModuleCache.get_toplevel_possibilities()`](https://github.com/python/mypy/blob/master/mypy/modulefinder.py#L162), but there's a [comment there](https://github.com/python/mypy/blob/master/mypy/modulefinder.py#L183-L185) which seems to say the observed behavior _shouldn't_ be happening.

```python
            # False positives are fine for correctness here, since we will check
            # precisely later, so we only look at the root of every filename without
            # any concern for the exact details.
```

**Comment by JukkaL** [comment]
I'm working on a fix. I'm hoping to release mypy 0.902 with the fix this week.

**Comment by JukkaL** [comment]
Fix is in #10609 (waiting for review). This seems specific to `google.*` packages. If you've encountered this issues with other packages, please leave a comment.

**Comment by kbrose** [comment]
Thanks for the prompt fix @JukkaL, I appreciate it!

**Comment by JukkaL** [comment]
he fix to this issue was released in mypy 0.902.

**Comment by alexanderdamiani** [comment]
I am still having the same issue with the `pyyaml` package after the update.

Environment: Docker
Base Image: windows-latest
`mypy` version: 0.902

```
error: Library stubs not installed for "yaml" (or incompatible with Python 3.8)  [import]
note: Hint: "python3 -m pip install types-PyYAML"
note: (or run "mypy --install-types" to install all missing stub packages)
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

**Comment by JukkaL** [comment]
@alexanderdamiani Your issue seems different. This issue about multiple libraries under the same top-level package, and I'm not aware other libraries that live under `yaml`. Please file a separate issue with more context.

**Comment by abhinavcoder** [comment]
@JukkaL 
I am facing a similar issue when using the following code:
```
from google.cloud import storage
```
Error:
```
error: Module "google.cloud" has no attribute "storage"
```

- Settings:
  - mypy: 0.942
  - OS: macOS
  - python: 3.7
I am running the mypy using the following command:
```
pipenv run mypy --install-types --non-interactive .
``` 
The above install the package `types-protobuf` which might be causing the issue.
